### DISCUSSION TITLE
[Question] Which library/framework have you been using together with jte?

### DISCUSSION THREAD

**Discussion by renannprado** [casid/jte#219] [Q&A]
I'm starting a new project, and I was wondering, from the users of this lib, what has been the go-to choice of library to pair with it.

Is it Svelte, htmx, react, jquery...?

Would be nice if you post your experience and tell why you used it, perhaps also some tips and tricks to make the most of it as well 😄.

If you have sample public repos, even better 😄!

@casid as a suggestion, maybe the "discussions" tab could be enabled? to avoid cluttering the issues which such posts.

**Comment by casid** [comment]
That's a good idea! I've enabled the discussions tab, ~~do you want to move this over there~~? Just found out it is possible to convert an issue to a discussion, that's neat!

**Comment by edward3h** [comment]
I've built a couple of projects using SvelteKit, but not in combination with jte. The benefit of SvelteKit is using the same templates for server and client. I don't want to mix jte with a different template language! At my day job we have a big React codebase and I kind of hate it.

One thing I am using with jte is [DaisyUI](https://daisyui.com/). It has components built with CSS on top of Tailwind. I was amazed at how much interactivity can be added with just CSS now. That project is not shareable now but hopefully soon.

On the backend I am using Micronaut for everything new. This should not be a surprise since I wrote the micronaut-views-jte integration 😁

**Reply by renannprado** [reply]
@edward3h thanks for the reply. I didn't know about DaisyUI before, I'll have a look!

When CSS/DaisyUI is not enough for you, don't you use something like htmx, at least? I haven't tried extensively, but I'm currently thinking of exploring at least the idea of using DaisyUI (now) + webcomponents + htmx + jte. For me, It's not about using yet another templating language but rather how to organize the codebase in a nice, maintainable way.

I also wanted to avoid having a npm/nodejs project attached to my build, but if I go down the route of using something like DaisyUI, then I think I have no choice.

If ever you find time to prepare a minimal example of your setup, with the build and everything, I would be happy to take a look at.

**Reply by edward3h** [reply]
[This is my build.gradle including the Tailwind/DaisyUI](https://github.com/edward3h/kv2/blob/tea-roster-parser/views/build.gradle) stuff. It uses [gradle node plugin](https://github.com/edward3h/kv2/blob/tea-roster-parser/views/build.gradle) to run the nodejs part. You can configure it to download nodejs so you don't have to have it installed, though I'm not doing that in this script.

I plan to use Hotwire/Turbo eventually since Micronaut Views includes Turbo support. Haven't got there yet though.

My only projects at work using jte are prototypes and internal tools, so I haven't needed any extra frontend tech.

**Reply by renannprado** [reply]
Ok, understood. I also would like to have a look at Hotwire/Turbo, but didn't get the chance yet.

Thanks a lot for the reply!
